src/insexpand.c

```diff
@@ -2012,17 +2017,10 @@ ins_compl_active(void)
  * Return True when wp is the actual completion window
  */
     int
-ins_compl_win_active(win_T *wp UNUSED)
+ins_compl_win_active(win_T *wp)
 {
-    return ins_compl_active()
-#if defined(FEAT_QUICKFIX)
-        && (!wp->w_p_pvw
-# ifdef FEAT_PROP_POPUP
-            && !(wp->w_popup_flags & POPF_INFO)
-# endif
-        )
-#endif
-    ;
+    return ins_compl_active() && wp == compl_curr_win
+	&& wp->w_buffer == compl_curr_buf;
```

Comment:
Shouldn't we keep the deleted part?

Reply:
It is no longer needed because we introduced a new variable.

Reply:
Are we sure, we cannot accidentally set completion in a popup window or previewwindow?

Reply:
e.g. when you do `setbufvar('%', '&previewwindow', 1)`?